eras/shelley-ma/test-suite/test/Test/Cardano/Ledger/Mary/Value.hs

```diff
@@ -354,6 +361,85 @@ compactRoundTrip = forAll gen $ \v ->
 compactTest :: TestTree
 compactTest = testProperty "fromCompact . toCompact == id" compactRoundTrip
 
+coinCBORTests :: TestTree
+coinCBORTests =
+  testGroup
+    "Coin CBOR roundtrips"
+    [ testCase "decode 0 as Coin for pre-conway" $
+        forM_ [natVersion @2 .. natVersion @8] $ \version ->
+          roundTripVersionExpectation version (Coin (0))
+    , testCase "fail decode 0 as Coin for conway" $
+        roundTripFailureExpectation (natVersion @9) (Coin (0))
+    , testCase "decode -1 as Coin for pre-conway" $
+        forM_ [natVersion @2 .. natVersion @8] $ \version ->
+          roundTripVersionExpectation version (Coin (-1))
```

Comment:
I always recommend switching to property instead of a test case whenever there is a chance for many potential inputs:
```suggestion
    , testProperty "decode -1 as Coin for pre-conway" $ $ \(Negative c) ->
          roundTripRanngeExpectation minBound (natVersion @8) (Coin c)
```

Try to go through all of the test and see what can be generated instead of hard coded